server.go

```diff
@@ -1325,12 +1326,41 @@ func newServer(cfg *Config, listenAddrs []net.Addr,
 		cfg.HealthChecks.DiskCheck.Attempts,
 	)
 
+	tlsHealthCheck := healthcheck.NewObservation(
+		"tls",
+		func() error {
+			_, parsedCert, err := cert.LoadCert(
+				cfg.TLSCertPath, cfg.TLSKeyPath,
+			)
+			if err != nil {
+				return err
+			}
+			outdated, err := cert.IsOutdated(
```

Comment:
`IsOutdated` checks the ip addresses + dns names on the cert, rather than its expiry. This can be replaced with a check on `cert.NotAfter` to detect expiry. 

I think we can leave `IsOutdated` out of this check entirely because we already have the `tlsautorefresh` flag which will sync the cert with our config at startup if enabled.

Reply:
Nice catch! I didn't even think to look at the source method 😅